runner/utils.py

```diff
@@ -521,57 +687,81 @@ def wait_for_job_and_check_result(batch_service_client: batch.BatchExtensionsCli
         check_stop_thread(stop_thread)
 
         if has_timedout(timeout_jobs_tasks_complete):
-            raise ex.JobTimedoutException(job_id, "Timed out waiting for job to complete.")
+            raise ex.JobTimedoutException(
+                job_id, "Timed out waiting for job to complete.")
 
         tasks = batch_service_client.task.list(job_id)
-        
-        incomplete_tasks = [task for task in tasks if task.state != batchmodels.TaskState.completed]
-        
-        failed_tasks = [task for task in tasks if task.execution_info.failure_info]
+
+        incomplete_tasks = [
+            task for task in tasks if task.state != batchmodels.TaskState.completed]
+
+        failed_tasks = [
+            task for task in tasks if task.execution_info.failure_info]
 
         if failed_tasks:
-            exception_message = " {} Task failure Errors: ".format(len(failed_tasks))
+            exception_message = " {} Task failure Errors: ".format(
+                len(failed_tasks))
             for failed_task in failed_tasks:
-                exception_message += "ErrorCode: {}. ErrorMessage: {}".format(failed_task.execution_info.failure_info.code, failed_task.execution_info.failure_info.message)
+                exception_message += "ErrorCode: {}. ErrorMessage: {}".format(
+                    failed_task.execution_info.failure_info.code, failed_task.execution_info.failure_info.message)
             raise ex.JobFailedException(job_id, exception_message)
-            
+
         if not incomplete_tasks:
             job_has_completed = True
             if not does_task_output_file_exist(batch_service_client, job_id, expected_output):
-                raise ex.JobFailedException(job_id, "Failed to find output {}".format(expected_output))
+                raise ex.JobFailedException(
+                    job_id, "Failed to find output {}".format(expected_output))
         else:
             logger.info("Job [{}] is running".format(job_id))
             time.sleep(5)
 
+
 def timedelta_since(start_time: datetime):
```

Comment:
doesn't this need a return type like -> datetime.datetime
like has_timedout below

Reply:
it has one?:
 :return: The timedelta between now and start_time
 :rtype: datetime.timedelta

Reply:
method needs a type hint https://docs.python.org/3/library/typing.html